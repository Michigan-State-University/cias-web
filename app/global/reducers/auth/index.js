export {
  LOG_IN_USER,
  LOG_OUT,
  REDIRECT_QUERY_KEY,
  ACCOUNT_CONFIRMATION_ERROR,
  ACCOUNT_CONFIRMATION_SUCCESS,
  RESET_REDUCER,
} from './constants';
export {
  logIn,
  logOut,
  editUserRequest,
  changePasswordRequest,
  changeErrorStatus,
  addAvatarRequest,
  deleteAvatarRequest,
  changeEmailRequest,
  changePhoneNumberRequest,
  confirmPhoneNumberRequest,
  sendSmsTokenRequest,
  logInGuestRequest,
  editPhoneNumberPreviewRequest,
  loginRequest,
  verificationCodeRequest,
} from './actions';
export { authReducer, initialState } from './reducer';
export {
  makeSelectAuth,
  makeSelectUser,
  makeSelectErrors,
  makeSelectLoaders,
  makeSelectUserRoles,
  makeSelectPhoneNumberPreview,
  makeSelectLoginFormData,
  makeSelectVerificationNeeded,
  makeSelectVerificationSuccess,
  makeSelectUserId,
} from './selectors';
export {
  logOutSaga,
  editUserSaga,
  changePasswordSaga,
  addAvatarSaga,
  deleteAvatarSaga,
  changeEmailSaga,
  changePhoneNumberSaga,
  confirmPhoneNumberSaga,
  sendSmsTokenSaga,
  editPhoneNumberQuestionSaga,
  loginSaga,
  verifyCodeSaga,
} from './sagas';
export { UserStorageController } from './UserStorageController';
