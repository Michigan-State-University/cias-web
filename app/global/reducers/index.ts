import { liveChatReducerKey } from './liveChat/reducer';
import { LiveChatState } from './liveChat/types';
import { chatWidgetReducerKey } from './chatWidget/reducer';
import { ChatWidgetState } from './chatWidget/types';
import { navigatorSetupReducerKey } from './navigatorSetup/reducer';
import { NavigatorSetupState } from './navigatorSetup/types';
import { notificationsReducerKey } from './notifications/reducer';
import { NotificationsState } from './notifications/types';

export type RootState = {
  [liveChatReducerKey]: LiveChatState;
  [chatWidgetReducerKey]: ChatWidgetState;
  [navigatorSetupReducerKey]: NavigatorSetupState;
  [notificationsReducerKey]: NotificationsState;
};
