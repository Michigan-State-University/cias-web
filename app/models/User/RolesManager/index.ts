export * from './defaultNavbarTabs';
export * from './useRoleManager';
export * from './UserRoles';
export * from './RolesColor';
export * from './navbarNames';
