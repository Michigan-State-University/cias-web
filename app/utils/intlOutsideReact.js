import { createIntl, createIntlCache } from 'react-intl';

import {
  DEFAULT_LOCALE,
  isAppLanguageSupported,
  translationMessages,
} from 'i18n';
import { store } from 'configureStore';

import { intlProviderConfig } from 'containers/AppLanguageProvider';

const cache = createIntlCache();

const intl = (locale = DEFAULT_LOCALE) =>
  createIntl(
    {
      locale: isAppLanguageSupported(locale) ? locale : DEFAULT_LOCALE,
      messages: translationMessages[locale],
      ...intlProviderConfig,
    },
    cache,
  );

export const formatMessage = (...props) => {
  const locale = store.getState()?.language?.locale;
  return intl(locale).formatMessage(...props);
};

export const formatInterventionLanguageMessage = (...props) => {
  const locale = store.getState()?.intervention?.intervention?.languageCode;
  return intl(locale).formatMessage(...props);
};

export default intl;
