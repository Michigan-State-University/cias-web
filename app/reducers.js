import { connectRouter } from 'connected-react-router';
import { combineReducers } from 'redux';

import history from 'utils/history';
import languageProviderReducer from 'containers/LanguageProvider/reducer';

import { authReducer } from 'global/reducers/auth';
import { globalStateReducer } from 'global/reducers/globalState';

/**
 * Merges the main reducer with the router state and dynamically injected reducers
 */
export default function createReducer(injectedReducers = {}) {
  const rootReducer = combineReducers({
    auth: authReducer,
    language: languageProviderReducer,
    router: connectRouter(history),
    global: globalStateReducer,
    ...injectedReducers,
  });

  return rootReducer;
}
