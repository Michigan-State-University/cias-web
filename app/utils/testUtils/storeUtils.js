import { createStore } from 'redux';
import createReducer from 'reducers';
import { initialState as authReducerInitialState } from 'global/reducers/auth/reducer';

export const createTestStore = (initialState) => {
  const reducer = (state) => state;

  const store = createStore(reducer, {
    auth: authReducerInitialState,
    ...initialState,
  });

  store.runSaga = () => {};
  store.injectedReducers = {};
  store.injectedSagas = {};
  store.createReducer = createReducer;

  return store;
};
