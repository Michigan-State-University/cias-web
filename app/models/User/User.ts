import { CamelToSnakeOmitId } from 'global/types/camelToSnake';
import { Roles } from './RolesManager/UserRoles';

export interface User {
  active: boolean;
  email: string;
  emailNotification: boolean;
  feedbackCompleted: boolean;
  firstName: string;
  fullName: string;
  id: string;
  lastName: string;
  phoneName: Nullable<string>;
  roles: Roles[];
  smsNotification: boolean;
  teamId: Nullable<string>;
  teamName: Nullable<string>;
}

export type UserDTO = CamelToSnakeOmitId<User>;
