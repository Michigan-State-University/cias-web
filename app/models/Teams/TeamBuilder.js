/* eslint-disable camelcase */
import { mapTeam } from 'utils/mapResponseObjects';
import Team from './Team';

export class TeamBuilder {
  /**
   * @param  {string} id
   * @returns  {TeamBuilder}
   */
  withId = (id) => {
    this.id = id;

    return this;
  };

  /**
   * @param  {string} name
   * @returns  {TeamBuilder}
   */
  withName = (name) => {
    this.name = name;

    return this;
  };

  /**
   * @param  {object} teamAdmin
   * @returns  {TeamBuilder}
   */
  withTeamAdmin = (teamAdmin) => {
    this.teamAdmin = teamAdmin;

    return this;
  };

  /**
   * @param  {object} json
   * @returns  {TeamBuilder}
   */
  fromJson = (json) => {
    const team = mapTeam(json);

    this.id = team.id;
    this.name = team.name;
    this.userId = team.userId;

    return this;
  };

  /**
   * @returns  {Team}
   */
  build = () =>
    new Team({
      id: this.id,
      name: this.name,
      teamAdmin: this.teamAdmin,
    });
}
