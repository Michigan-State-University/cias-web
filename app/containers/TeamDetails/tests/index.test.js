/**
 *
 * Tests for TeamDetails
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { DEFAULT_LOCALE } from 'i18n';

import { initialState as authState } from 'global/reducers/auth';
import { createTestStore } from 'utils/testUtils/storeUtils';
import { TeamBuilder } from 'models/Teams/TeamBuilder';
import intlOutsideReact from 'utils/intlOutsideReact';
import { Roles } from 'models/User/RolesManager';
import { TeamDetails } from '../index';

describe('<TeamDetails />', () => {
  let props;
  const initialState = {
    auth: {
      ...authState,
      user: {
        id: '1',
        roles: [Roles.admin],
      },
    },
    teamList: {
      singleTeam: new TeamBuilder()
        .withId('ta-1')
        .withName('Team 1')
        .withTeamAdmin({ name: 'TA 1', email: 'ta-1@test.pl', id: 'id-ta-1' })
        .build(),
      loaders: { singleTeamLoading: false },
      errors: { singleTeamError: null },
    },
  };
  let store;

  beforeEach(() => {
    store = createTestStore(initialState);

    props = {
      user: initialState.auth.user,
      teamList: initialState.teamList,
      match: {
        params: {
          id: 'ta-1',
        },
      },
      fetchTeam: jest.fn(),
      intl: intlOutsideReact,
    };
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <MemoryRouter>
            <TeamDetails {...props} />
          </MemoryRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <MemoryRouter>
            <TeamDetails {...props} />
          </MemoryRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
