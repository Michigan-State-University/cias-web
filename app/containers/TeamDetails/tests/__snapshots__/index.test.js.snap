// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamDetails /> Should render and match the snapshot 1`] = `
<div>
  <div
    class="Box-sc-1vhutan-0 styled__StyledBox-sc-1x1cot6-0 bdpEXp jKbXJq"
    height="100%"
    width="100%"
  >
    <a
      class="styled__StyledLink-sc-3pirdo-0 jceFfF"
      href="/teams"
    >
      <img
        alt="traingle"
        class="Img-sc-55j3su-0 gfzPuQ"
        src="IMAGE_MOCK"
      />
      <p
        class="Text-e4p0w3-0 ikSWkc"
        color="#C866EA"
        font-weight="bold"
      >
        <span>
          Back to Team List
        </span>
      </p>
    </a>
    <div
      style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: center; justify-content: flex-start;"
    >
      <div
        style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: auto; flex-basis: auto; flex-grow: 0; flex-shrink: 0; max-width: 100%; margin-left: 0%;"
      >
        <h1
          class="H1-sc-1k7m23t-0 eMTspO"
        >
          <span>
            Team Details
          </span>
        </h1>
      </div>
      <div
        style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: auto; flex-basis: auto; flex-grow: 0; flex-shrink: 0; max-width: 100%; margin-left: 0%;"
      >
        <button
          class="StyledTextButton-sc-1b7egxw-0 bsIeDJ"
          color="#C866EA"
          font-weight="bold"
        >
          <span>
            + Invite to team
          </span>
        </button>
      </div>
    </div>
    <div
      style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;"
    >
      <div
        style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;"
      >
        <div
          class="Box-sc-1vhutan-0 styled__CardBox-sc-1x1cot6-3 cVZZaQ jfWIRU"
        >
          <h2
            class="H2-uw8lbl-0 iUhbVg"
          >
            <span>
              Edit Team
            </span>
          </h2>
          <div
            style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;"
          >
            <div
              style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 50%; flex-basis: 50%; flex-grow: 0; flex-shrink: 0; max-width: 50%; margin-left: 0%;"
            >
              <p
                class="Text-e4p0w3-0 cKQscp"
                font-size="14"
              >
                <span>
                  Change Team name
                </span>
              </p>
              <input
                class="Input-t9j2q5-0 kyuVel"
                placeholder="e.g Team n.o. 1"
                type="text"
                value="Team 1"
                width="100%"
              />
            </div>
            <div
              style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 50%; flex-basis: 50%; flex-grow: 0; flex-shrink: 0; max-width: 50%; margin-left: 0%;"
            >
              <p
                class="Text-e4p0w3-0 moOet"
                font-size="14"
              >
                <span>
                  Change Team Admin
                </span>
              </p>
              <div
                class="Box-sc-1vhutan-0 cVZZaQ"
              >
                <div
                  class="styled__InlineLoader-sc-1aaagww-1 kkTeLU"
                  data-testid="inline-loader"
                >
                  <div
                    class="styled__StyledSpinner-sc-129uixn-0 OJHNY"
                    color="#C866EA"
                    size="50"
                    width="6"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;"
          >
            <div
              style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;"
            >
              <button
                class="StyledButton-d2s0gt-0 ihkoZy"
                color="primary"
                disabled=""
                font-size="14"
                font-weight="bold"
                width="200"
              >
                <span>
                  Save
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start; margin-top: 75px;"
    >
      <div
        style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;"
      >
        <div
          class="styled__FillScreenLoader-sc-1aaagww-0 bcnRTW"
          data-testid="fullscreen-loader"
          opacity="0.5"
        >
          <div
            class="styled__StyledSpinner-sc-129uixn-0 iMjPBI"
            color="#C866EA"
            size="100"
            width="6"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
